Sources/SWBBuildService/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+#[[
+This source file is part of the Swift open source project
+
+Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See http://swift.org/LICENSE.txt for license information
+See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+]]
+
+add_library(SWBBuildService
+  BuildDependencyInfo.swift
+  BuildOperationMessages.swift
+  BuildService.swift
+  BuildServiceEntryPoint.swift
+  ClientExchangeDelegate.swift
+  DependencyGraphMessages.swift
+  DocumentationInfo.swift
+  LocalizationInfo.swift
+  Messages.swift
+  PlanningOperation.swift
+  PreviewInfo.swift
+  ProjectDescriptor.swift
+  Session.swift
+  Tools.swift)
```

Comment:
Any chance to use glob instead of manually listing all sources? Current approach introduces a lot of code to maintain, and extra work when files are added or renamed.

Reply:
No, that doesn't work as you would expect. The glob is applied at configure time and so any changes after the initial configuration would be ignored.

Reply:
No, globbing source files does not work and frequently causes issues with files being mysteriously omitted from the library: https://discourse.cmake.org/t/cmake-doesnt-register-new-files-when-i-add-new-files/13390

Updating the CMakeLists.txt when you add a file is part of the incremental build flow, invalidating that dependency.

Reply:
Thanks for the answer.
Small question.
Is this list present (or will be) in any other configs nearby?
Should it have a separate single source of truth that is imported by all other build systems?

Reply:
Cmake documentation mentions `CONFIGURE_DEPENDS` that was added in 3.12 to make globing work correctly. Although it may depend on generator. https://cmake.org/cmake/help/latest/command/file.html

Reply:
Agreed. If there is no known other use for these lists they can stay as is.